### ISSUE TITLE
Support configuration of the `lifecycle` image

### ISSUE THREAD

**Issue by braunsonm** [knative/func#2184]
When you run a build with the `func` CLI using your own builder it will attempt to pull the lifecycle binaries from `quay.io`. This may be because the builder is not [trusted](https://buildpacks.io/docs/tools/pack/concepts/trusted_builders/)

However it doesn't seem that the `func` CLI respects the list of trusted builders that a user configures with the Pack CLI, and also does not allow configuration of this property itself. This means that at runtime `func` will attempt to pull the lifecycle from `quay.io`. The image that `func` pulls is _hardcoded_ here: https://github.com/knative/func/blob/51f7a70b9a750f0509b79940a3c9628696ffc688/pkg/builders/buildpacks/builder.go#L117

This should allow configuration, or allow configuration of trusted builders so the built in lifecycle binaries can be used.

**Comment by lkingland** [comment]
This seems like a bug to me.  I will mark it as such.

It's perhaps worth mentioning that the builder lifecycle image is not intended as "hard-coded" (meaning an unchangable string built into the binary), but rather it is _defaulted_.  This should be an optional user-modifiable setting.

**Comment by braunsonm** [comment]
I agree @lkingland however does `func` plan to follow the trusted builder methodology I linked from CNB? If so it should just use the included lifecycle binaries if the builder is trusted.

**Comment by EraKin575** [comment]
I would like to work on this issue!
